core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -272,6 +274,10 @@ class Partition(val topic: String,
       if (logManager.getLog(topicPartition, isFuture = true).isDefined)
         logManager.asyncDelete(topicPartition, isFuture = true)
     }
+    // If the deleted replica was leader replica, there might be delayed produce requests in purgatory need to be cleaned up.
```

Comment:
Nit: Looks like delayed fetch and delete records will be cleaned up as well, not just delayed produce requests.